### ISSUE TITLE
Where is a full list of the mtk# classes that the editor uses? (I've already asked on Stack Overflow and received no answers. I assume this is too niche)

### ISSUE THREAD

**Issue by thedonquixotic** [Microsoft/vscode#51956]
I'm using a custom css loader extension to add some extra font styling to my editor's theme. I can search through the divs etc using the VScode dev tools, but I was wondering if there might be some documentation that was more comprehensive? The names of the classes are things like mtk3, mtk16, mtk2 etc. I can sort of guess what some of these classes are being assigned based on but it would be much easier to customize my theme if I could just read documentation. 

Also, now that Microsoft has bought Github, why don't ya'll just bring over the css theme editing stuff from Atom? That would make this a whole lot easier!

**Comment by aeschli** [comment]
We don't support styling on the DOM level. The DOM structure and clases on it (`mtk`) are implementation details of VSCode. We want to have the freedom to change them at any time.

To change styles (colors, fonts, font styles), we support themes and user settings. See https://code.visualstudio.com/docs/getstarted/themes#_customizing-a-color-theme for instructions.
If we lack a specific theming capability, please file a feature request

**Comment by thedonquixotic** [comment]
@aeschli Hey yeah, I know about the themes customizing options, and I also realize that you guys expect to be changing your classes often but even so I would find it useful to see a list, and/or if nothing else it seems that this would be information that was in the VScode documentation somewhere. VScode is open source right? So if someone wanted to contribute to the code their would be someplace they could go to see what the current classes were and what they could do with them, correct?

Again, I know about the theme customizing options but what I am doing with my custom CSS is in fact not possible to do with the theme system.

**Comment by aeschli** [comment]
For development question, the code is the ultimate documentation. We have some developer wikis but you have to understand that we focus on documenting things that are designed to be stable and maintained. We're also happy to help through the various channels.

The class names are actually dynamically generated based on the currently chosen token theme. See [here](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/modes/supports/tokenization.ts#L387).
It's really not designed for what you are aiming to do.
Please help us to make the theming capabilities more powerful. What do you want to add with your custom CSS?

**Comment by thedonquixotic** [comment]
@aeschli  I'll gladly give my perspective and experience. I'm really more of a designer than coder so it's a bit hard for me to directly contribute through code but hopefully my experiments can help!

There's a bunch of stuff that I'd like to be able to do more with my theming. The main thing that I'm using the custom CSS for is to change font-families. While this may seem odd, putting my comments in san serif proportional fonts and the rest of my code in a monospace font makes things much easier to read. 

![image](https://user-images.githubusercontent.com/9451726/41709436-c2aa4d6a-7508-11e8-8c4e-8702813d41cb.png)


I've also used a regex extension quite a bit. Some of these things wouldn't make any sense for you guys to program into VScode and are really more conventions on my part. For instance, most of my theme is a nice desaturated warm purple (based on the Hopscotch theme), with for areas with `// comment` a nice dull grey. Areas that are ``//- comment" however are bright yellow with dark black text. This way I can use these notation for setting up section titles and other places I want to be more visible. 

![image](https://user-images.githubusercontent.com/9451726/41709495-e7a133a4-7508-11e8-99fa-5c6decdfb7df.png)


The other very big thing I'd like to do, and I've literally spent like 6 hours today trying to figure out is how to properly highlight my html tags. In particular tags which are being used in Angular templates etc and therefore might not be standard HTML named tags or have multiple carriage returns and or spaces inside their tags. I've got an extension that can select the brackets that you click on and that's super useful but 1) I'd like to be able to style all my matched brackets not just the one I've clicked on and 2) the extension allows you to style the opening bracket, interior of the tag, and the closing bracket, but it doesn't differentiate between the closing and opening tag which is a problem because I'd like to be able to style a border on the left side of the opening bracket of the opening tag, and a border on the right side of the closing bracket of the closing tag, that way I can use the styling to kind of visually "encase" the tag's contents.
